### ISSUE TITLE
[BUG] xr.concat inverts coordinates order

### ISSUE THREAD

**Issue by clausmichele** [pydata/xarray#4072]
Following the issue #3969 
Merging two datasets using xr.concat inverts the coordinates order.

#### MCVE Code Sample

```
import numpy as np
import xarray as xr

x = np.arange(0,10)
y = np.arange(0,10)
time = [0,1]
data = np.zeros((10,10), dtype=bool)
dataArray1 = xr.DataArray([data], coords={'time': [time[0]], 'y': y, 'x': x},
                             dims=['time', 'y', 'x'])
dataArray2 = xr.DataArray([data], coords={'time': [time[1]], 'y': y, 'x': x},
                             dims=['time', 'y', 'x'])
dataArray1 = dataArray1.to_dataset(name='data')
dataArray2 = dataArray2.to_dataset(name='data')

print(dataArray1)
print(xr.concat([dataArray1,dataArray2], dim='time'))
```
#### Current Output

```
<xarray.Dataset>
Dimensions:  (time: 1, x: 10, y: 10)
Coordinates:
  * time     (time) int64 0
  * y        (y) int64 0 1 2 3 4 5 6 7 8 9
  * x        (x) int64 0 1 2 3 4 5 6 7 8 9
Data variables:
    data     (time, y, x) bool False False False False ... False False False
<xarray.Dataset>
Dimensions:  (time: 2, x: 10, y: 10)
Coordinates:
  * x        (x) int64 0 1 2 3 4 5 6 7 8 9  ##Inverted x and y
  * y        (y) int64 0 1 2 3 4 5 6 7 8 9
  * time     (time) int64 0 1
Data variables:
    data     (time, y, x) bool False False False False ... False False False

```

#### Expected Output
```
<xarray.Dataset>
Dimensions:  (time: 1, x: 10, y: 10)
Coordinates:
  * time     (time) int64 0
  * y        (y) int64 0 1 2 3 4 5 6 7 8 9
  * x        (x) int64 0 1 2 3 4 5 6 7 8 9
Data variables:
    data     (time, y, x) bool False False False False ... False False False
<xarray.Dataset>
Dimensions:  (time: 2, x: 10, y: 10)
Coordinates:
  * y        (y) int64 0 1 2 3 4 5 6 7 8 9
  * x        (x) int64 0 1 2 3 4 5 6 7 8 9
  * time     (time) int64 0 1
Data variables:
    data     (time, y, x) bool False False False False ... False False False

```

#### Problem Description

The concat function should not invert the coordinates but maintain the original order.

#### Versions

<details><summary>INSTALLED VERSIONS
</summary>
------------------
commit: None
python: 3.6.8 (default, May  7 2019, 14:58:50)
[GCC 8.3.0]
python-bits: 64
OS: Linux
OS-release: 4.15.0-88-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: C.UTF-8
LOCALE: en_US.UTF-8
libhdf5: None
libnetcdf: None

xarray: 0.15.1
pandas: 1.0.3
numpy: 1.18.2
scipy: None
netCDF4: None
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: None
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: None
distributed: None
matplotlib: 3.2.0
cartopy: None
seaborn: None
numbagg: None
setuptools: 46.1.3
pip: 9.0.1
conda: None
pytest: None
IPython: 7.13.0
sphinx: 2.4.3</details>

**Comment by keewis** [comment]
don't use the `Coordinates` section to check the order. `Dataset` objects may have multiple data variables (each with a possibly different order of dimensions) so for displaying it needs to define some kind of order.

What you want to compare is the order of dimensions referenced by a variable:
```
<xarray.Dataset>
Dimensions:  (time: 1, x: 10, y: 10)
Coordinates:
  * time     (time) int64 0
  * y        (y) int64 0 1 2 3 4 5 6 7 8 9
  * x        (x) int64 0 1 2 3 4 5 6 7 8 9
Data variables:
    data     (time, y, x) bool False False False False ... False False False
             ^^^^^^^^^^^^
```
and you will notice that the order does not change.

To make this less confusing, maybe we should always sort the coordinates when displaying them? I'm guessing that right now they are displayed in the order they were added to the coordinates.

**Comment by kmuehlbauer** [comment]
This rings a bell! It has be discussed in https://github.com/pydata/xarray/issues/2811. I've done extensive checks back then, but didn't come up with an PR.

My workaround or solution is outlined [in this comment](https://github.com/pydata/xarray/issues/2811#issuecomment-473810333). Code might have been changed!

**Comment by clausmichele** [comment]
@keewis yes you are right, but still a consistent ordering would be less confusing, including maybe also the Dimensions field.
Now Dimensions shows first x and then y, even though the data itself has y,x ordering.

**Comment by keewis** [comment]
sorry, I forgot to remove the "closes" item. This will be fixed by #4419.

**Comment by keewis** [comment]
closed by #4419
